### PR TITLE
Add API for using torque to iob/RobotHardware

### PR DESCRIPTION
**Pull request by YoheiKakiuchi**
* add setServoTorqueGainPercentage to RobotHardwareService

add new iob API
* read_torque_pgain
* write_torque_pgain
* read_torque_dgain
* write_torque_dgain

* add setJointControlMode to RobotHardwareService
add implementation of write_control_mode in iob

### CODE REVIEW

**Comment by YoheiKakiuchi** [comment]
> if you changed API, that should be v3?
I think so.
This PR and https://github.com/fkanehiro/hrpsys-base/pull/1246, https://github.com/fkanehiro/hrpsys-base/pull/1248 should be v3.

**Comment by fkanehiro** [comment]
It is not clear (not defined) when we should increment IOB_VERSION.
To my understanding, V1 has bee used to keep compatibility (mainly for Nextage Open) and V2 was used for new development so far.
Do we need now fix V2?

**Comment by k-okada** [comment]
I believe everytime we update the iob API, then we should increment the
IOV_VERSION.
Please see example below, I think ask them to update their source tree is
not a good solution, they should be able to continue building their source
without any changes, even if we update the iob of latest hrpsys.

```
JAXON,CHIDORI向けの環境構築READMEに従って環境設定をしていたのですが、hrpsys_trans_bridgeのビルドで以下のようなエラーが起きました。

Errors << hrpsys_trans_bridge:make
/home/leus/catkin_ws/jaxon_tutorial/logs/hrpsys_trans_bridge/build.make.012.log
CMakeFiles/RobotHardwareComp_trans.dir/home/leus/catkin_ws/jaxon_tutorial/devel/share/hrpsys/src/rtc/RobotHardware/robot.cpp.o:
関数 robot::setJointInertia(char const*, double)' 内:
robot.cpp:(.text+0x3570):write_joint_inertia' に対する定義されていない参照です
CMakeFiles/RobotHardwareComp_trans.dir/home/leus/catkin_ws/jaxon_tutorial/devel/share/hrpsys/src/rtc/RobotHardware/robot.cpp.o:
関数 robot::writeAccelerationCommands(double const*)' 内:
robot.cpp:(.text+0x2094):write_command_accelerations' に対する定義されていない参照です
CMakeFiles/RobotHardwareComp_trans.dir/home/leus/catkin_ws/jaxon_tutorial/devel/share/hrpsys/src/rtc/RobotHardware/robot.cpp.o:
関数 robot::setJointInertias(double const*)' 内:
robot.cpp:(.text+0x35d4):write_joint_inertias' に対する定義されていない参照です
```

--
◉ Kei Okada


2018-06-13 14:16 GMT+09:00 Fumio KANEHIRO <notifications@github.com>:

**Comment by fkanehiro** [comment]
Yes, I know that such thing will happen if I add a new API. But I didn't know that there are people who are using V2 except us...
Ok then, let's increment every time we add new API.
#1246,#1248 will go to V3 and this PR go to V4?

**Comment by k-okada** [comment]
that's sound good

--
◉ Kei Okada


2018-06-13 19:29 GMT+09:00 Fumio KANEHIRO <notifications@github.com>:

**Comment by fkanehiro** [comment]
@YoheiKakiuchi 
Could you move your APIs to V4?

**Comment by YoheiKakiuchi** [comment]
#1246,#1248 have been moved to V3 and this PR has been moved to V4

File: rtc/RobotHardware/robot.cpp [hunk 1]
```diff
@@ -107,15 +118,40 @@ bool robot::loadGain()
...
+                sstrm >> dummy;
+                if(sstrm.eof()) {
+                    default_tqpgain[i] = default_tqdgain[i] = 0.0;
+                } else {
+                    sstrm >> default_tqpgain[i];
+                    sstrm >> dummy;
+                    sstrm >> default_tqdgain[i];
+                }
+            } else {
+                std::cerr << "[RobotHardware] error: exist empty line in gain file" << std::endl;
```
Comment (fkanehiro) [inline rtc/RobotHardware/robot.cpp]: 関節数が多いロボットでは部位ごとに空行をはさんだりしていたのですが、空行が挟まっているとマズイ場面がありますでしょうか？ @YoheiKakiuchi

Comment (YoheiKakiuchi) [reply rtc/RobotHardware/robot.cpp]: すいません。
以下で対応しました。
https://github.com/fkanehiro/hrpsys-base/pull/1252

**Pull request merged**